core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceUnitTestsSpec.scala

```diff
@@ -122,8 +143,125 @@ class MultiCloudWorkspaceServiceUnitTestsSpec
+    when(requestContext.otelContext).thenReturn(None)
+    val service = spy(
+      MultiCloudWorkspaceService.constructor(
+        mock[SlickDataSource],
+        mock[WorkspaceManagerDAO],
+        mock[BillingProfileManagerDAO],
+        mock[SamDAO],
+        mock[MultiCloudWorkspaceConfig],
+        mock[LeonardoDAO],
+        "MultiCloudWorkspaceService-test"
+      )(requestContext)
+    )
 
+    val workspaceService = spy(
+      WorkspaceService.constructor(
+        dataSource = mock[SlickDataSource],
+        methodRepoDAO = mock[MethodRepoDAO],
+        cromiamDAO = mock[ExecutionServiceDAO],
+        executionServiceCluster = mock[ExecutionServiceCluster],
+        execServiceBatchSize = 10,
+        workspaceManagerDAO = mock[WorkspaceManagerDAO],
+        leonardoService = mock[LeonardoService],
+        methodConfigResolver = mock[MethodConfigResolver],
+        gcsDAO = mock[GoogleServicesDAO],
+        samDAO = mock[SamDAO],
+        notificationDAO = mock[NotificationDAO],
+        userServiceConstructor = _ => mock[UserService],
+        genomicsServiceConstructor = _ => mock[GenomicsService],
+        maxActiveWorkflowsTotal = 1000,
+        maxActiveWorkflowsPerUser = 100,
+        workbenchMetricBaseName = "test",
+        submissionCostService = mock[SubmissionCostService],
+        config = mock[WorkspaceServiceConfig],
+        requesterPaysSetupService = mock[RequesterPaysSetupService],
+        entityManager = mock[EntityManager],
+        resourceBufferService = mock[ResourceBufferService],
+        servicePerimeterService = mock[ServicePerimeterService],
+        googleIamDao = mock[GoogleIamDAO],
+        terraBillingProjectOwnerRole = "terraBillingProjectOwnerRole",
+        terraWorkspaceCanComputeRole = "terraWorkspaceCanComputeRole",
+        terraWorkspaceNextflowRole = "terraWorkspaceNextflowRole",
+        terraBucketReaderRole = "terraBucketReaderRole",
+        terraBucketWriterRole = "terraBucketWriterRole",
+        rawlsWorkspaceAclManager = mock[RawlsWorkspaceAclManager],
+        multiCloudWorkspaceAclManager = mock[MultiCloudWorkspaceAclManager],
+        fastPassServiceConstructor = (_, _) => mock[FastPassService]
+      )(requestContext)
+    )
+
+    doReturn(Future(sourceWorkspace))
+      .when(service)
+      .getV2WorkspaceContextAndPermissions(sourceWorkspace.toWorkspaceName, SamWorkspaceActions.read, None)
+
+    val billingProject = RawlsBillingProject(
+      RawlsBillingProjectName(destWorkspaceRequest.namespace),
+      CreationStatuses.Ready,
+      None,
+      None
+    )
+
+    doReturn(Future(billingProject))
+      .when(service)
+      .getBillingProjectContext(RawlsBillingProjectName(destWorkspaceRequest.namespace), requestContext)
+
+    doReturn(Future()).when(service).requireCreateWorkspaceAction(billingProject.projectName, requestContext)
+
+    // Because there is no billing profile, we will call the legacy GCP clone method.
```

Comment:
Just to clarify, this calls the legacy GCP clone method because it's cloning a Rawls workspace, not because there's a missing billing profile right? We have Rawls workspaces with billing profiles now

Reply:
Good point. We call the legacy method if either there is no billing profile or there is one, but its cloudPlatform is GCP. I will change the mocking to return a GCP billing profile so it is clearer and testing our new expected path.

```
        case (RawlsWorkspace, profileOpt)
            if profileOpt.isEmpty ||
              profileOpt.map(_.getCloudPlatform).contains(CloudPlatform.GCP) =>